test/unit/org/apache/cassandra/index/sai/plan/OperationTest.java

```diff
@@ -0,0 +1,478 @@
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        protected String toString(boolean cql)
+        {
+            AbstractType<?> type = column.type;
+            switch (operator)
+            {
+                case CONTAINS:
+                    assert type instanceof CollectionType;
+                    CollectionType<?> ct = (CollectionType<?>)type;
+                    type = ct.kind == CollectionType.Kind.SET ? ct.nameComparator() : ct.valueComparator();
+                    break;
+                case CONTAINS_KEY:
+                    assert type instanceof MapType;
+                    type = ((MapType<?, ?>)type).nameComparator();
+                    break;
+                case IN:
+                    type = ListType.getInstance(type, false);
+                    break;
+                default:
+                    break;
+            }
+            return cql
+                   ? String.format("%s %s %s", column.name.toCQLString(), operator, type.toCQLString(value) )
+                   : String.format("%s %s %s", column.name.toString(), operator, type.getString(value));
+        }
+    }
+
+    private static DecoratedKey buildKey(Object... key) {
+        return buildKey(BACKEND, key);
+    }
+
+    private static DecoratedKey buildKey(ColumnFamilyStore cfs, Object... key) {
+        AbstractType<?> type = cfs.metadata().partitionKeyType;
+        ByteBuffer decomposed;
+        if(type instanceof CompositeType)
+        {
+            Preconditions.checkArgument(key.length == type.subTypes().size());
+            decomposed = ((CompositeType) type).decompose(key);
+        }
+        else
+        {
+            Preconditions.checkArgument(key.length == 1);
+            decomposed = ((AbstractType) type).decompose(key[0]);
+        }
+        return Murmur3Partitioner.instance.decorateKey(decomposed);
+    }
+
+    private static Unfiltered buildRow(Cell... cells)
+    {
+        return buildRow(Clustering.EMPTY, null, cells);
+    }
+
+    private static Row buildRow(Row.Deletion deletion, Cell... cells)
+    {
+        return buildRow(Clustering.EMPTY, deletion, cells);
+    }
+
+    private static Row buildRow(Clustering clustering, Cell... cells)
+    {
+        return buildRow(clustering, null, cells);
+    }
+
+    private static Row buildRow(Clustering clustering, Row.Deletion deletion, Cell... cells)
+    {
+        Row.Builder rowBuilder = BTreeRow.sortedBuilder();
+        rowBuilder.newRow(clustering);
+        for (Cell c : cells)
+            rowBuilder.addCell(c);
+
+        if (deletion != null)
+            rowBuilder.addRowDeletion(deletion);
+
+        return rowBuilder.build();
+    }
+
+    private static Cell buildCell(ColumnMetadata column, ByteBuffer value, long timestamp)
+    {
+        return BufferCell.live(column, timestamp, value);
+    }
+
+    private static Cell deletedCell(ColumnMetadata column, long timestamp, int nowInSeconds)
```

Comment:
nit: Pretty much all the `Cell` and `Clustering` types between here and 433 could be `Cell<?>` and `Clustering<?>`